src/components/screens/forms/form-history-screen.tsx

```diff
@@ -1,12 +1,198 @@
-import { Typography } from "@mui/material";
+import React, { useContext, useEffect, useState } from "react";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import { NavigationTabContainer } from "styled/layouts/navigations";
+import NavigationTab from "components/layouts/navigations/navigation-tab";
 import strings from "localization/strings";
-import React from "react";
+import { AdminFormListStack, AdminFormTypographyField } from "styled/react-components/react-components";
+import { useApiClient } from "app/hooks";
+import Api from "api";
+import { AuditLogEntry, MetaformMember } from "generated/client";
+import { ErrorContext } from "components/contexts/error-handler";
+import { useNavigate, useParams } from "react-router-dom";
+import FormRestrictedContent from "components/containers/form-restricted-content";
+import moment from "moment";
+import LocalizationUtils from "utils/localization-utils";
 
 /**
  * Form history screen component
  */
-const FormHistoryScreen: React.FC = () => (
-  <Typography>{ strings.generic.notImplemented }</Typography>
-);
+const FormHistoryScreen: React.FC = () => {
+  const errorContext = useContext(ErrorContext);
+
+  const apiClient = useApiClient(Api.getApiClient);
+  const { metaformMembersApi, auditLogEntriesApi, metaformsApi } = apiClient;
+  const [ auditLogEntries, setAuditLogEntries ] = useState<AuditLogEntry[]>([]);
+  const [ metaformMembers, setMetaformMembers ] = useState<MetaformMember[]>([]);
+  const [ loading, setLoading ] = useState(false);
+  const navigate = useNavigate();
+  const { formSlug } = useParams();
+
+  /**
+   * Loads audit log entries
+   *
+   * @param metaformId metaform id
+   */
+  const loadAuditLogEntries = async (metaformId: string) => {
+    try {
+      // TODO pagination maybe???? Gets heavier as time goes now 35000 entries.
```

Comment:
This was discussed and for the time being we shall leave the API response as it is.

Reply:
I would say probably after the MVP has released since the logs tends to get huge but yeah we can leave it as it is for now